.github/workflows/python.yml

```diff
@@ -169,12 +167,23 @@ jobs:
         run: |
           echo "::set-output name=python-version::$(awk 'FNR == 2' .python-version)"
 
-
       # Until GH composite actions can use `uses`, we need to setup python here
       - uses: actions/setup-python@v2
         with:
           python-version: ${{ steps.python-version.outputs.python-version }}
 
+      - name: Setup pip
+        uses: ./.github/actions/setup-pip
+        id: pip
+
+      - name: pip cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip.outputs.pip-cache-dir }}
+          key: ${{ runner.os }}-pip-py3-${{ hashFiles('**/requirements-*.txt') }}
```

Comment:
Changed the key so that it doesn't clash with py2 cache

Reply:
Very nice.